Добавлена реализация метода разделения файла на части

AntonZinovik opened the pull request:
Program:
 - добавлено подключение класса сервиса.

Добавлено:
- контроллер для взаимодействия с файлом FileController;
- сервис с методом разделения файла на части и архивация частей файла; 
- настройки размера буффера и директория для сохранения частей файлов вынесены в appsettings;

Удаленно:
- тестовый контроллер

AntonZinovik committed:
Добавлена реализация метода разделения файла на части

AntonZinovik committed:
Исправление комментариев

mBereberdin commented on readme.md:
```diff
@@ -0,0 +1,5 @@
+
+[![Build](https://github.com/AntonZinovik/FileTransferService/actions/workflows/Build.yml/badge.svg?branch=main)](https://github.com/AntonZinovik/FileTransferService/actions/workflows/Build.yml)
+
+Сервис отвечающий за раздение большого файла на части и отправку данных частей.
```
раздение - разделение

mBereberdin commented on src/FileTransferService/Controllers/FileController.cs:
```diff
@@ -0,0 +1,54 @@
+/// </summary>
+[ApiController]
+[Route("[controller]")]
+public class FileController : ControllerBase
```
FilesController

mBereberdin commented on src/FileTransferService/Controllers/FileController.cs:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /// <summary>
+    /// Разделеие файла на части.
```
Разделеие - разделение

mBereberdin commented on src/FileTransferService/Controllers/FileController.cs:
```diff
@@ -0,0 +1,54 @@
+    /// <summary>
+    /// Разделеие файла на части.
+    /// </summary>
+    /// <param name="filePath">Логгер.</param>
```
Точно логгер?

mBereberdin commented on src/FileTransferService/Controllers/FileController.cs:
```diff
@@ -0,0 +1,54 @@
+    /// <param name="cancellationToken">Токен отмены выполнения операции.</param>
+    /// <exception cref="ArgumentNullException">Когда путь до файла оказался пустым.</exception>
+    /// <response code="204">Когда удалось разделить файл.</response>
+    [HttpPost("File")]
```
File - split

mBereberdin commented on src/FileTransferService/Controllers/FileController.cs:
```diff
@@ -0,0 +1,54 @@
+
+        if (string.IsNullOrEmpty(filePath))
+        {
+            throw new ArgumentNullException(filePath);
```
Можно сообщение написать, что параметр не корректный

mBereberdin commented on src/FileTransferService/Options/FilesOptions.cs:
```diff
@@ -0,0 +1,26 @@
+    /// <param name="pathChunkDirectory">Директория для сохранения частей файла.</param>
+    public FilesOptions(int bufferSize, string pathChunkDirectory)
+    {
+        PathChunkDirectory = pathChunkDirectory;
```
Валидацию на пустую или из пробелов?

mBereberdin replied:
Ниже и сколько места выделено - можно проверить. Если там будет меньше или 0 - будут ошибки.
И вообще можно тип с инта на uint поменять, чтобы отрицательного точно не было и только на 0 проверять

mBereberdin commented on src/FileTransferService/Program.cs:
```diff
@@ -25,8 +30,11 @@
-    .Configuration(context.Configuration));
+                                                                 .Configuration(context.Configuration));
+
+builder.Services.Configure<FilesOptions>(builder.Configuration.GetSection("FileOptions"));
```
FileOptions -> nameof(FileOptions)

mBereberdin commented on src/FileTransferService/Services/Interfaces/IFileService.cs:
```diff
@@ -0,0 +1,14 @@
+    /// </summary>
+    /// <param name="filePath">Путь до файла.</param>
+    /// <param name="cancellationToken">Токен отмены выполнения операции.</param>
+    Task SplitFile(string filePath, CancellationToken cancellationToken);
```
SplitFileAsync

mBereberdin commented on src/FileTransferService/Services/Implementations/FileService.cs:
```diff
@@ -0,0 +1,64 @@
+
+            await using var targetStream = File.Create(nameChunkFile);
+            await using var compressionStream = new GZipStream(targetStream, CompressionMode.Compress);
+            await compressionStream.WriteAsync(buffer.AsMemory(0, bytesRead), cancellationToken);
```
buffer.AsMemory(0, bytesRead) - предложил бы тоже в переменную вынести, мало ли сколько у него вызовов, чтобы стек не тянуть

AntonZinovik committed:
Исправлены замечания

- Исправлена орфография комментариев;
- Добавлена проверка настроек файлов;
- Исправлено наименование метода в FileService;
- исправлено наименование контроллера FilesController;

mBereberdin commented on src/FileTransferService/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,54 @@
+
+        if (string.IsNullOrEmpty(filePath))
+        {
+            throw new ArgumentNullException(filePath,"Параметр не корректный");
```
Пробел между параметром и строкой
